tests/mismatch.cpp

```diff
@@ -0,0 +1,244 @@
+
+  auto expected = std::mismatch(v.begin(), v.end(), v.begin());
+
+  sycl::sycl_execution_policy<class MismatchAlgorithmEqualOneRange> snp{};
+  auto actual = std::experimental::parallel::mismatch(
+      snp, v.begin(), v.end(), v.begin(), v.end(),
+      [](int a, int b) { return a == b; });
+
+  EXPECT_EQ(actual, expected);
+}
+
+TEST_F(MismatchAlgorithm, TestMismatchMatchOneRangeCustomPredicate) {
+  std::vector<int> v1{0, 1, 2, 3, 4, 5, 6, 7};
+  std::vector<int> v2{1, 2, 3, 4, 5, 6, 7, 8};
+
+  auto expected = std::mismatch(v1.begin(), v1.end(), v2.begin(),
+      [](int a, int b) { return a < b; });
+
+  sycl::sycl_execution_policy<class MismatchAlgorithmMatchOneRangeCustomPredicate> snp{};
+  auto actual = std::experimental::parallel::mismatch(
+      snp, v1.begin(), v1.end(), v2.begin(),
+      [](int a, int b) { return a < b; });
+
+  EXPECT_EQ(actual, expected);
+}
+
+TEST_F(MismatchAlgorithm, TestMismatchNoMatchOneRangeCustomPredicate) {
+  std::vector<int> v1{0, 1, 9, 3, 4, 5, 6, 7};
+  std::vector<int> v2{1, 2, 3, 4, 5, 6, 7, 8};
+
+  auto expected = std::mismatch(v1.begin(), v1.end(), v2.begin(),
+      [](int a, int b) { return a < b; });
+
+  sycl::sycl_execution_policy<class MismatchAlgorithmNoMatchOneRangeCustomPredicate> snp{};
+  auto actual = std::experimental::parallel::mismatch(
+      snp, v1.begin(), v1.end(), v2.begin(),
+      [](int a, int b) { return a < b; });
+
+  EXPECT_EQ(actual, expected);
+}
+
+TEST_F(MismatchAlgorithm, TestMismatchMatchOneRangeNoPredicate) {
+  std::vector<int> v{0, 1, 2, 3, 4, 5, 6, 7};
+
+  auto expected = std::mismatch(v.begin(), v.end(), v.begin());
+
+  sycl::sycl_execution_policy<class MismatchAlgorithmMatchNoPredicate> snp{};
+  auto actual = std::experimental::parallel::mismatch(
+      snp, v.begin(), v.end(), v.begin());
+
+  EXPECT_EQ(actual, expected);
+}
+
+TEST_F(MismatchAlgorithm, TestMismatchEqualNoPredicate) {
+  std::vector<int> v{0, 1, 2, 3, 4, 5, 6, 7};
+
+  auto expected = std::mismatch(v.begin(), v.end(), v.begin());
+
+  sycl::sycl_execution_policy<class MismatchAlgorithmEqualNoPredicate> snp{};
+  auto actual = std::experimental::parallel::mismatch(
+      snp, v.begin(), v.end(), v.begin(), v.end());
+
+  EXPECT_EQ(actual, expected);
+}
+
+TEST_F(MismatchAlgorithm, TestMismatchEqualFirstSmaller) {
+  std::vector<int> v{0, 1, 2, 3, 4, 5, 6, 7};
+
+  auto expected = std::mismatch(v.begin(), v.end() - 2, v.begin());
+
+  sycl::sycl_execution_policy<class MismatchAlgorithmEqualFirstSmaller> snp{};
+  auto actual = std::experimental::parallel::mismatch(
+      snp, v.begin(), v.end() - 2, v.begin(), v.end(),
+      [](int a, int b) { return a == b; });
+
+  EXPECT_EQ(actual, expected);
+}
+
+TEST_F(MismatchAlgorithm, TestMismatchEqualSecondSmaller) {
+  std::vector<int> v{0, 1, 2, 3, 4, 5, 6, 7};
+
+  auto expected = std::mismatch(v.begin(), v.end() - 2, v.begin());
+  std::swap(std::get<0>(expected), std::get<1>(expected));
```

Comment:
```cpp
using std::swap;
swap(...);
```

(With the Ranges TS, you can just do `ranges::swap(...)`, but that's due to CPOs).